src/main/scala/it/pagopa/interop/authorizationprocess/common/readmodel/ReadModelQueries.scala

```diff
@@ -12,18 +11,19 @@ import org.mongodb.scala.model.Sorts.ascending
 
 import scala.concurrent.{ExecutionContext, Future}
 import java.util.UUID
+import spray.json._
 
 object ReadModelQueries {
 
-  def listClients(
+  def listClients[A: JsonReader](
```

Comment:
may I suggest this (awful name, feel free to peek another)?
```suggestion
  private def listGenericClients[A: JsonReader](
```
and create
```
def listClients(...): Future[PaginatedResult[PersistentClient]]
```
and 
```
def listClientsWithKeys(): Future[PaginatedResult[ReadModelClientWithKeys]]
```
to not delegate outside the schema stored on mongodb